lua/lazyvim/plugins/extras/editor/neogen.lua

```diff
@@ -0,0 +1,45 @@
+return {
+  { -- documentation generator
+    "danymat/neogen",
+    opts = { enabled = true },
+    keys = {
+      {
+        "<leader>cnc",
+        function()
+          require("neogen").generate({ type = "class" })
+        end,
+        desc = "Generate documentation for class",
+      },
+      {
+        "<leader>cnf",
+        function()
+          require("neogen").generate({ type = "func" })
+        end,
+        desc = "Generate documentation for function",
+      },
+      {
+        "<leader>cnF",
+        function()
+          require("neogen").generate({ type = "file" })
+        end,
+        desc = "Generate documentation for file",
+      },
+      {
+        "<leader>cnt",
+        function()
+          require("neogen").generate({ type = "type" })
+        end,
+        desc = "Generate documentation for type",
+      },
+    },
+  },
+
+  { -- which-key integration
+    "folke/which-key.nvim",
+    opts = {
```

Comment:
Probably want to guard it here with `optional = true`. Don't wanna bind this Extra to `which-key` explicitly I think, in case someone is using something else.